crates/orderbook/src/order_validation.rs

```diff
@@ -497,6 +507,16 @@ async fn get_quote_and_check_fee(
     Ok(quote)
 }
 
+/// Checks whether or not an order is a market order.
+///
+/// A market order is defined as an order whose price is equal or worse to the
+/// best on-chain price that can be found at the time. This allows the protocol
+/// to detect orders that place orders that aren't intended to be traded right
+/// away and flag them as liquidity orders.
+fn is_market_priced_order(order: &OrderData, quote: &Quote) -> bool {
+    order.sell_amount.full_mul(quote.buy_amount) >= quote.sell_amount.full_mul(order.buy_amount)
+}
```

Comment:
This should already be checked at this point.

Agree that we would check if this were a public function, but since I just moved it out for improved readability and testability, I think its fine just to look at the data it cares about.